packages/rs-sdk/src/sync.rs

```diff
@@ -187,21 +190,91 @@ where
         async move {
             let settings = closure_settings.load_full().clone();
             let mut func = inner_fn.lock().await;
-            (*func)(*settings).await
+            let result = (*func)(*settings).await;
+
+            // Ban or unban the address based on the result
+            match &result {
+                Ok(response) => {
+                    // Unban the address if it was banned and node responded successfully this time
```

Comment:
Not sure if it will happen - it should be already unbanned by lower layer (but we can keep it here as a precaution)